Samples/Desktop/D3D12HelloWorld/src/HelloVADecode/D3D12HelloVADecode.cpp

```diff
@@ -0,0 +1,634 @@
+    ThrowIfFailed(va_status, "vaQueryConfigEntrypoints for VAProfileNone");
+
+    bool supportsVideoProcessing = false;
+    for (int32_t i = 0; !supportsVideoProcessing && i < num_entrypoints; i++) {
+        if (entrypoints[i] == VAEntrypointVideoProc)
+            supportsVideoProcessing = true;
+    }
+
+    if (!supportsVideoProcessing) {
+        ThrowIfFailed(VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT, "VAEntrypointVideoProc not supported.");
+    }
+}
+
+void D3D12HelloVADecode::InitVAProcContext()
+{
+    VAStatus va_status = vaCreateConfig(
+        m_vaDisplay,
+        VAProfileNone,
+        VAEntrypointVideoProc,
+        nullptr,
+        0,
+        &m_VAProcConfigId);
+    ThrowIfFailed(va_status, "vaCreateConfig");
+
+    va_status = vaCreateContext(
+        m_vaDisplay,
+        m_VAProcConfigId,
+        GetWidth(),
+        GetHeight(),
+        VA_PROGRESSIVE,
+        m_VARenderTargets,
+        FrameCount,
+        &m_vaColorConvCtx);
+    ThrowIfFailed(va_status, "vaCreateContext");
+        
+    va_status = vaCreateBuffer(
+        m_vaDisplay,
+        m_vaColorConvCtx,
+        VAProcPipelineParameterBufferType,
+        sizeof(VAProcPipelineParameterBuffer),
+        1,
+        NULL,
+        &m_vaColorConvBuf);
+    ThrowIfFailed(va_status, "vaCreateBuffer");
+}
+
+void D3D12HelloVADecode::ImportRenderTargetsToVA()
+{
+    VASurfaceAttrib createSurfacesAttribList[3] = {
+        {
+            VASurfaceAttribPixelFormat,
+            VA_SURFACE_ATTRIB_SETTABLE,
+            {
+                VAGenericValueTypeInteger,
+                // Based on the default render target
+                // format DXGI_FORMAT_R8G8B8A8_UNORM
+                VA_FOURCC_RGBA,
+            },
+        },
+        {
+            VASurfaceAttribMemoryType,
+            VA_SURFACE_ATTRIB_SETTABLE,
+            {
+                VAGenericValueTypeInteger,
+                // Using NTHandles for interop is the safe way of sharing resources between the 
+                // VADisplay D3D12 device and the app/sample D3D12 independent devices
+                VA_SURFACE_ATTRIB_MEM_TYPE_NTHANDLE,
+            },
+        },
+        {
+            VASurfaceAttribExternalBufferDescriptor,
+            VA_SURFACE_ATTRIB_SETTABLE,
+            {
+                VAGenericValueTypePointer,
+                // createSurfacesAttribList[2].value.value.p is set in code below
+                0,
+            },
+        },
+    };
+
+    // The value here is an array of num_surfaces pointers to HANDLE so
+    // each handle can be associated with the corresponding output surface
+    // in the call to vaCreateSurfaces
+    HANDLE renderTargets[FrameCount];
+    for (size_t i = 0; i < FrameCount; i++)
+    {
+        HRESULT hr = m_device->CreateSharedHandle(m_renderTargets[i].Get(),
+            nullptr,
+            GENERIC_ALL,
+            nullptr,
+            &renderTargets[i]);
+        ThrowIfFailed(hr);
+        assert(renderTargets[i]);
```

Comment:
Why have this? Are we worried CreateSharedHandle won't work?

Reply:
Yeah, now that you point it out it's a bit redundant as we check for HR success for CreateSharedHandle right above.